wc3-hw1-TranNgocLam/cypress/pages/registrationForm.js

```diff
@@ -0,0 +1,279 @@
+// import chaiColors from 'chai-colors';
+
+// import tinycolor from 'tinycolor2';
+export const registerForm = {
+    TXT_LASTNAME: "#lastName",
+    TXT_FIRSTNAME: "#firstName",
+    RDO_GENDER: "#genterWrapper input",
+    LBL_GENDER: "#genterWrapper label",
+    TXT_EMAIL: "#userEmail",
+    TXT_USERNUMBER: "#userNumber",
+    BTN_SUBMIT: "#submit",
+    DTP_DOB: "#dateOfBirthInput",
+    DTP_DOB_DAY: ".react-datepicker__day--0",
+    DTP_DOB_MONTH: ".react-datepicker__month-select",
+    DTP_DOB_YEAR: ".react-datepicker__year-select",
+    DDL_SUBJECTS: "#subjectsInput",
+    CHK_HOBBIES: "#hobbiesWrapper :checkbox",
+    DLG_UPLOADPICTURE: "#uploadPicture",
+    TXT_CURRENTADDRESS: "#currentAddress",
+    DDL_STATE: "#state input",
+    DDL_CITY: "#city input",
+    RED: "rgb(220, 53, 69)",
+    GREEN: "rgb(40, 167, 69)",
+
+    inputLastName(lastName) {
+        cy.get(this.TXT_LASTNAME).type(lastName, { force: true });
+        return this;
+    },
+
+    inputFirstName(firstName) {
+        cy.get(this.TXT_FIRSTNAME).type(firstName, { force: true });
+        return this;
+    },
+
+    inputEmail(email) {
+        cy.get(this.TXT_EMAIL).type(email, { force: true });
+        return this;
+    },
+    checkGender(gender) {
+        cy.get(this.RDO_GENDER).check(gender, {force: true} )
+        return this;
+    },
+
+    inputUserNumber(userNumber) {
+        cy.get(this.TXT_USERNUMBER).type(userNumber);
+        return this;
+    },
+
+    chooseDOB(day, month, year) {
+        // Open date picker
+        cy.get(this.DTP_DOB).click();
+        // Select year
+        cy.get(this.DTP_DOB_YEAR).select(year);
+        // Select month
+        cy.get(this.DTP_DOB_MONTH).select(month);
+        // Select day
+        cy.get(this.DTP_DOB_DAY + day).first().click();
+        return this
+    },
+
+    inputSubjects(subjects) {
+        for (let i = 0; i < subjects.length; i++) {
+            cy.get(this.DDL_SUBJECTS).type(subjects[i]).type('{enter}')
+        }
+
+        return this;
+    },
+
+    checkHobbies(hobbies) {
+        for (let i = 0; i < hobbies.length; i++) {
+            switch (hobbies[i]) {
```

Comment:
What happen when there are 100 checkboxes for 100 hobbies? Make your code more general.
Hint: are you recognize that you can select checkboxes by clicking on their label? You can use this:
```js
hobbies.forEach((hobby) => {
      cy.get(locatore).contains(hobby).click();
});
```